c10/core/ScalarType.h

```diff
@@ -512,6 +512,19 @@ static inline ScalarType promoteTypes(ScalarType a, ScalarType b) {
     return ScalarType::Undefined;
   }
 
+  // To avoid 7 rows/cols consiting only of Undefined promotion
+  // related to qints and bits between bfloat16 and float8 dtypes,
```

Comment:
I'm confused by this. The logic above ensures that if either a or b is qint or bits, we never reach here. So we don't need to add such entries in the matrix.

Reply:
Float8 dtypes have enum values 23 and 24 (`static_cast<int>`), which should be aligned 1 to 1 with `_promoteTypesLookup` table.
If we don't want to add to this table 7 undefined rows matching dtypes between bfloat (enum 16) and float8 (23/24), we have to substract 7 from float8 enums when indexing promotion table to avoid out of range error:
`return _promoteTypesLookup[static_cast<int>(a)][static_cast<int>(b)];`

Reply:
Hoooo
Ok, I did not understood that from your comment!
Could you update the comment to say that the Bits and Quantized dtypes are 7 dtypes starting at position 17 (If I calculated properly) and are already handled and not including in the table below. So every value above them needs to be shifted down to account for that.
And make the if check anything > 18 (so that it will also handle future dtypes we will add).

Reply:
Good point. I have updated the comment and "unhardcoded" number 7 to the distance between specific dtypes.